models/userSchema.js

```diff
@@ -0,0 +1,66 @@
+const mongoose = require('mongoose');
+const bcrypt = require('bcryptjs');
+const validator = require('validator');
+const userSchema = new mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+      maxLength: 22,
+    },
+    email: {
+      type: String,
+      required: true,
+      unique: true,
+      trim: true,
+      lowercase: true,
+      validate: [validator.isEmail, 'Provide valid Email Address'],
+    },
+    contact: {
+      type: Number,
+      maxLength: 12,
+      trim: true,
+      validate: [validator.isInt, 'Provide Numbers Only'],
+      callingCode: {
+        type: mongoose.ObjectId,
+        ref: 'CallingCode',
+      },
+    },
+    department: {
+      type: String,
+      required: true,
+    },
+    role: {
+      type: String,
+      enum: ['STUDENT', 'TEACHER'],
+      required: true,
+    },
+    id: {
+      type: Number,
+      unique: true,
+      required: true,
+    },
+    // TODO -  WE HAVE TO DECIDE SHOULD WE WANT TO SAVE PASSWORD IN DB OR NOT?
+    // * DONE
+    password: {
+      type: String,
+      required: true,
+      validate: [validator.isAscii, 'Provide Valid Password'],
+    },
+    confirmPassword: {
```

Comment:
Confirm password is also not needed, since we use it on frontend to double check if student has correctly written the password or not.